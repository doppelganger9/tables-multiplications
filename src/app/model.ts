export type Action = 'RÃ©viser' | 'Afficher';

export interface StatistiqueReponses {
  min: number;
  max: number;
  moy: number;
  vuesSurLEnsemble: number;
  reponsesCorrectesSurLEnsemble: number;
  reponses: Array<Reponse>;
}

// pour une question de type X fois Y = Z, trouver Y
export interface Reponse {
  tempsMillisecondes: number;
  correcte: boolean;
  nombre: number;
  operande: number;
  reponse: number;
}

export interface Question {
  nombre: number;
  operande: number;
  total?: number;
  dateDebut: Date;
  finie: boolean;
}
