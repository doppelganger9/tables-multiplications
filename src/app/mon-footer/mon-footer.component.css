div {
  width: fit-content;
  margin: auto;
  font-size: small;
}