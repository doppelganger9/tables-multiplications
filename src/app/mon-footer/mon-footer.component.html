<div data-e2e="version-container">
  <span
    title="Version {{ (version$ | async)?.version }} - {{
      (version$ | async)?.shortSHA
    }} - {{ (version$ | async)?.lastCommitTime | date: 'full' }} "
    >v{{ (version$ | async)?.version }}</span
  >
  &nbsp;-&nbsp;
  <span
    ><a href="https://github.com/doppelganger9/tables-multiplications"
      >sources</a
    ></span
  >
</div>
