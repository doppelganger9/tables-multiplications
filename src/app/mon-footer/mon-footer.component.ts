import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs';
import { StateService } from '../store/state.service';
import { VersionData } from '../model';
import { DatePipe } from '@angular/common';

@Component({
  selector: 'app-footer',
  templateUrl: './mon-footer.component.html',
  styleUrls: ['./mon-footer.component.css'],
  providers: [DatePipe],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class MonFooterComponent implements OnInit {
  version$: Observable<VersionData>;

  constructor(private stateService: StateService) {}

  ngOnInit(): void {
    this.version$ = this.stateService.getVersion();
  }
}
