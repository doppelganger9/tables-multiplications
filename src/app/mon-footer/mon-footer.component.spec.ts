import { MonFooterComponent } from './mon-footer.component';

import { StateService } from '../store/state.service';
import { TestBed } from '@angular/core/testing';
import { of } from 'rxjs';
import { VersionData } from '../model';

describe(`MonFooterComponent`, () => {
  beforeEach(async () => {
    TestBed.configureTestingModule({
      imports: [],
      providers: [
        {
          provide: StateService,
          useValue: {
            getVersion: () =>
              of({
                version: 'test',
                shortSHA: 'fake',
                lastCommitTime: new Date(
                  Date.parse('2023-10-18T20:52:16.000+0200')
                )
              } as VersionData)
          } as Partial<StateService>
        }
      ],
      declarations: [MonFooterComponent]
    }).compileComponents();
  });

  it('should create the component', () => {
    const fixture = TestBed.createComponent(MonFooterComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it('should render the version from stateService', () => {
    const fixture = TestBed.createComponent(MonFooterComponent);

    fixture.detectChanges();

    expect(
      (fixture.debugElement.nativeElement as HTMLElement).querySelector(
        `[data-e2e="version-container"]`
      )
    ).toMatchSnapshot();
  });
});
