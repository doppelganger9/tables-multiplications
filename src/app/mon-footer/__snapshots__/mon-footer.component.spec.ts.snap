// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonFooterComponent should render the version from stateService 1`] = `
<div
  data-e2e="version-container"
>
  <span
    title="Version test - fake - mercredi 18 octobre 2023 à 20:52:16 GMT+02:00 "
  >
    vtest
  </span>
    -  
  <span>
    <a
      href="https://github.com/doppelganger9/tables-multiplications"
    >
      sources
    </a>
  </span>
</div>
`;
